ci/blackbox/rdsbroker_test.go

```diff
@@ -436,6 +436,58 @@ var _ = Describe("RDS Broker Daemon", func() {
 				"postgres-micro-without-snapshot-10",
 			)
 		})
+	})
+
+	Describe("complex plan upgrade failure and recovery", func() {
+		TestUpdatePlan := func(serviceID, startPlanID, upgradeToPlanID, extensionName, expectedAwsTagPlanID, fixPlanID string) {
+			var (
+				instanceID string
+			)
+
+			BeforeEach(func() {
+				instanceID = uuid.NewV4().String()
+
+				brokerAPIClient.AcceptsIncomplete = true
+
+				code, operation, err := brokerAPIClient.ProvisionInstance(instanceID, serviceID, startPlanID, fmt.Sprintf(`{"enable_extensions": ["%s"]}`, extensionName))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("succeeded"))
+			})
+
+			AfterEach(func() {
+				brokerAPIClient.AcceptsIncomplete = true
+				code, operation, err := brokerAPIClient.DeprovisionInstance(instanceID, serviceID, startPlanID)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("gone"))
+			})
+
+			It("handles a failure to update to a plan with a newer engine version", func() {
+				code, operation, err := brokerAPIClient.UpdateInstance(instanceID, serviceID, startPlanID, upgradeToPlanID, `{}`)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(code).To(Equal(202))
+				state := pollForOperationCompletion(brokerAPIClient, instanceID, serviceID, startPlanID, operation)
+				Expect(state).To(Equal("failed"))
```

Comment:
Are we still returning a message "please contact support" at this point? I guess if they're still unable to upgrade to 11 we should still be telling them that, even though they can be moved back "on plan" without our intervention.

(Arguably, if we now define "on plan" as "having *at least* the right about of disk space..." they never move *off*-plan in this scenario...)

Reply:
In this situation the user would get the message "Plan upgrade failed. Refer to database logs for more information.". I think that error is fine.

I've updated the wording around the tests.